### PR TITLE
Integration

### PR DESCRIPTION
**Pull request by Fahim-Khan-P**
### **We added the following:**

- [ ] Before N+1 Problem:

before

- [ ] After Fixing N+1:
after

- Use Capybara to write integration tests for each view in your project.

- [ ] Think on the flow of things and group the tests of page in a way that makes sense. For example, one test would check for the things in the page and another test for clicking links on it.

- [ ] User index page:
1. I can see the username of all other users.
2. I can see the profile picture for each user.
3. I can see the number of posts each user has written.
4. When I click on a user, I am redirected to that user's show page.

- [ ] user show page:
1. I can see the user's profile picture.
2. I can see the user's username.
3. I can see the number of posts the user has written.
4. I can see the user's bio.
5. I can see the user's first 3 posts.
6. I can see a button that lets me view all of a user's posts.
7. When I click a user's post, it redirects me to that post's show page.
8. When I click to see all posts, it redirects me to the user's post's index page.

- [ ] User post index page:
1. I can see the user's profile picture.
2. I can see the user's username.
3. I can see the number of posts the user has written.
4. I can see a post's title.
5. I can see some of the post's body.
6. I can see the first comments on a post.
7. I can see how many comments a post has.
8. I can see how many likes a post has.
9. I can see a section for pagination if there are more posts than fit on the view.
10. When I click on a post, it redirects me to that post's show page.

- [ ] Post show page:
1. I can see the post's title.
2. I can see who wrote the post.
3. I can see how many comments it has.
4. I can see how many likes it has.
5. I can see the post body.
6. I can see the username of each commentor.
7. I can see the comment each commentor left.

### CODE REVIEW

File: spec/features/users_spec.rb [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+      expect(page).to have_content(@user5.posts_counter)
+    end
+    it 'Shows the link to each user' do
+      visit users_path
+      expect(page).to have_link(@user.name, href: user_path(@user))
+      expect(page).to have_link(@user2.name, href: user_path(@user2))
+      expect(page).to have_link(@user3.name, href: user_path(@user3))
+      expect(page).to have_link(@user4.name, href: user_path(@user4))
+      expect(page).to have_link(@user5.name, href: user_path(@user5))
+    end
```
Comment (codepantha) [inline spec/features/users_spec.rb:45]: - [ ] You are currently asserting that a link to go to the user's show page exists. However, what is required is that you write tests that redirects to the user's show page when you click on a user.

image

File: spec/features/posts_spec.rb [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+      @comment2 = Comment.create!(text: 'This is a comment2', author_id: @user.id, post_id: @post.id)
+      @comment3 = Comment.create!(text: 'This is a comment3', author_id: @user.id, post_id: @post.id)
+      visit user_posts_path(@user)
+    end
+    it 'displays the user name' do
+      expect(page).to have_content(@user.name)
+    end
+    it 'displays the user posts counter' do
+      expect(page).to have_content(@user.posts_counter)
+    end
```
Comment (codepantha) [inline spec/features/posts_spec.rb]: - [ ] Please add a test that asserts you can see the user's profile picture. This is a requirement 👍

**Review by codepantha — CHANGES_REQUESTED** [review]
# STATUS: CHANGES REQUIRED ♻️ 

## Hi team 👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights 🎯 
✔️ Descriptive PR
✔️ Added integration tests 📌 
✔️ Linters are passing

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- n/a

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by omarsalem7 — APPROVED** [review]
Hi @Fahim-Khan-P 😊, 
# STATUS: APPROVED! 🥇 💯 



Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 🚀 
Congratulations! 🎉 

## Highlights
✅ Project has met all the technical aspects you needed for this project. Keep rocking! 

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**